backend/dataall/cdkproxy/stacks/pivot_role.py

```diff
@@ -512,105 +396,31 @@ def _create_dataall_policy1(self, env_resource_prefix: str) -> iam.ManagedPolicy
                     actions=[
```

Comment:
I am having a look at it and you are right, we do not tag the resource share so this policy should give an access denied when trying to update or delete the resource share request. I am doing some testing as this might be a bug in shares

Reply:
I have tested the creation and cleanup of a share request with the tag condition in the pivot role policy. The request is not affected by this condition and the pivotRole is able to delete the RAM request that is not tagged. In the [docs](https://docs.aws.amazon.com/service-authorization/latest/reference/list_awsresourceaccessmanagerram.html#awsresourceaccessmanagerram-resources-for-iam-policies) this condition is applicable to the resource-share, but in the past I also faced issues with some AWS services and the type of tag that they where checking (notice that there is a mismatch in the docs, in the resource types it does not include `ram:ResourceTag`).

 If we want to enforce the condition, we can try to modify it to check `ram:ResourceTag/${TagKey}` instead of `aws:ResourceTag/${TagKey}`. I just tested that, I modified the condition in `CreateResourceShare` and `DeleteResourceShare` and everything works fine without the resource share being tagged. I will remove the condition to avoid confusion and reach out to the RAM team to see if they can provide some guidance on this. What do you think?